lib/scsi_defs.h

```diff
@@ -0,0 +1,154 @@
+#ifndef LIB_SCSI_DEFS_H
+#define LIB_SCSI_DEFS_H
+
+namespace scsi_defs {
+
+
+  // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 2.4.1 Table 11
+  enum SenseResponse {
+    CURRENT_DESCRIPTOR_ERROR = 0x72,
+    DEFERRED_DESCRIPTOR_ERROR = 0x73,
+    CURRENT_FIXED_ERROR = 0x70,
+    DEFERRED_FIXED_ERROR = 0x71
+  };
+
+
+  // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 2.4.1.5 Table 28
+  enum SenseKey {
+    NO_SENSE = 0x0,
+    RECOVERED_ERROR = 0x1,
+    NOT_READY = 0x2,
+    MEDIUM_ERROR = 0x3,
+    HARDWARE_ERROR = 0x4,
+    ILLEGAL_REQUEST = 0x5,
+    UNIT_ATTENTION = 0x6,
+    DATA_PROTECT = 0x7,
+    BLANK_CHECK = 0x8,
+    VENDOR_SPECIFIC = 0x9,
+    COPY_ABORTED = 0xA,
+    ABORTED_COMMAND = 0xB,
+    RESERVED = 0xC,
+    VOLUME_OVERFLOW = 0xD,
+    MISCOMPARE = 0xE,
+    COMPLETED = 0xF
+  };
+
+} // namepsace scsi_defs
+
+
+  enum Status {
+    GOOD = 0x0,
+    CHECK_CONDITION = 0x2,
+    CONDITION_MET = 0x4,
+    BUSY = 0x8,
+    INTERMEDIATE = 0x10,               // obsolete
+    INTERMEDIATE_CONDITION_MET = 0x14, // obsolete
+    RESERVATION_CONFLICT = 0x18,
+    COMMAND_TERMINATED = 0x22,         // obsolete
+    TASK_SET_FULL = 0x28,
+    ACA_ACTIVE = 0x30,
+    TASK_ABORTED = 0x40,
+  };
+
+  enum PeripheralDeviceType {
+    DIRECT_ACCESS_BLOCK = 0x0,
+    SEQUENTIAL_ACCESS = 0x1,
+    PRINTER = 0x2,
+    PROCESSOR = 0x3,
+    WRITE_ONCE = 0x4,
+    CD_DVD = 0x5,
+    OPTICAL_MEMORY = 0x7,
+    MEDIUM_CHANGER = 0x8,
+    STORAGE_ARRAY_CONTROLLER = 0xc,
+    ENCLOSURE_SERVICES = 0xd,
+    SIMPLIFIED_DIRECT_ACCESS = 0xe,
+    OPTICAL_CARD_READER_WRITER = 0xf,
+    BRIDGE_CONTROLLER_COMMANDS = 0x10,
+    OBJECT_BASED_STORAGE = 0x11,
+    AUTOMATION_DRIVE_INTERFACE = 0x12,
+    WELL_KNOWN_LOGICAL_UNIT = 0x1e,
+    UNKNOWN = 0x1f,
+  };
+
+  enum Version { // Field code of implemented version of the SPC standard
+    NO_STANDARD = 0x0,
+    SPC = 0x3,
+    SPC2 = 0x4,
+    SPC3 = 0x5,
+    SPC4 = 0x6,
+    SPC6 = 0x7,
+  };
+
+  enum OptCode { // Opt codes defined for SCSI commands supported by this project; this list may increase
+    REPORT_LUNS = 0xa0,
+    REQUEST_SENSE = 0x3,
+    INQUIRY = 0x12,
+    READ_CAPACITY_10 = 0x25,
+    SERVICE_ACTION_IN = 0x9e,
+    MODE_SENSE_6 = 0x1a,
+    MAINTENANCE_IN = 0xa3,
+    READ_TOC = 0x43,
+    DO_PREVENT_ALLOW_MEDIUM_REMOVAL = 0x1e,
+    PERSISTENT_RESERVE_IN = 0x5e,
+    PERSISTENT_RESERVE_OUT = 0x5f,
+    RESERVE_6 = 0x16,
+    RELEASE_6 = 0x17,
+    START_STOP_UNIT = 0x1b,
+    TEST_UNIT_READY = 0x0,
+  };
+
+  class ControlByte {
+   private:
+    uint8_t obsolete : 2;
+    uint8_t naca : 1;
+    uint8_t reserved : 3;
+    uint8_t vendorSpecific : 2;
+
+   public:
+    ControlByte() :
```

Comment:
ctors are not required - zero initialization can be used for first one and second is redundant if we can use aggregate initialization (https://en.cppreference.com/w/cpp/language/aggregate_initialization). We can also make all the fields public.

Reply:
Same as https://github.com/googleinterns/scsi2nvme/pull/17#discussion_r426740888